src/UI/EditorView.re

```diff
@@ -22,16 +22,28 @@ module Styles = {
   ];
 };
 
+let onFileDropped = (event: NodeEvents.fileDropEventParams) =>
+  List.iter(
+    path => {
+      let stats = Unix.stat(path);
+      // This tests if the path dropped is a file. We can't handle opening directories, etc. yet
+      if (stats.st_kind == S_REG) {
+        GlobalContext.current().dispatch(
+          Model.Actions.OpenFileByPath(path, None, None),
+        );
+      };
```

Comment:
side-effects should not be triggered in the UI, as that makes it difficult to track and control what's happening,a nd to evolve the logic when we need to update the state as well, for example. In short, it elads to spaghetti code. The rule of thumb is that the UI should only dispatch messages to be handled by stores/update functions, which then update the model and emit side-effects through `Isolinear.Effect`.

But even more generally, even in imperative-land, doing blocking I/O on the UI thread is considered a big no-no as it can blocks the UI from rendering and in some cases can really hurt responsiveness.

Reply:
Yeah that's my bad -- I kinda just quickly worked on this to get it to somewhat work. I should have added a WIP tag on the PR generally.

Thanks for the info!